internal/api/v1/user/configs/configs.go

```diff
@@ -0,0 +1,53 @@
+package configs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// route defines the API endpoint path for accessing configurations in the SPV Wallet API.
+// This constant is appended to the base URL to form the complete path for configuration
+// requests.
+const route = "api/v1/configs"
+
+// HTTPGetter is an interface that defines methods for making HTTP GET request.
+type HTTPGetter interface {
+	Get(ctx context.Context, path string) ([]byte, error)
+}
+
+// API represents a client for interacting with an API at a specific base path,
+// using an HTTPGetter interface for making requests.
+type API struct {
+	addr string     // addr defines the base URL for the user configurations API.
+	cli  HTTPGetter // cli is used for intializing the external SPV Wallet API HTTP calls.
+}
+
+// SharedConfig fetches the shared configuration from the SPV Wallet API.
+// It constructs the request path and unmarshals the response into a SharedConfig struct.
+func (a *API) SharedConfig(ctx context.Context) (*response.SharedConfig, error) {
+	path := a.addr + "/shared"
+	body, err := a.cli.Get(ctx, path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get shared config from %s: %w", path, err)
+	}
+
+	var dst response.SharedConfig
+	if err := json.Unmarshal(body, &dst); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal HTTP response into SharedConfig: %w", err)
+	}
```

Comment:
The reason ``resty`` is not used directly here is because the facade design pattern allows us to abstract away the use of a third-party library, providing a much simpler and elegant interface that we control ourselves.

However, there is potential a risk that after updating packages version by dependabot many of those 'direct' method calls can be inappropriate because of some major changes (for ex. removing some parameters, chaining their order etc.). Instead of updating them directly in many places, we'll update them only in the single place. 

Another aspect of this is the shared state of the resty HTTP client between multiple API implementations. Applying interface definitely narrows the access scope to the methods that can modify the state between the APIs in a way that can lead to some unexpected behaviors.